src/compas/datastructures/mesh/trimesh_samplepoints.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+from numpy import array
+from numpy.random import choice
+from numpy.random import rand
+from numpy import sqrt
+from numpy import float64
+from numpy import cross
+from numpy.linalg import norm
+from numpy import clip
+from numpy import finfo
+
+
+__all__ = [
+    'trimesh_samplepoints',
+]
+
+
+def trimesh_samplepoints(mesh, num_points: int = 1000, return_normals: bool = False):
+    """Compute sample points on a triangle mesh surface
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+        Mesh is limited to triangle mesh
+    num_points : (int)
+        How many points sampled
+    return_normals : (bool)
+        if True, return the normals vector of sampled points
+
+    Returns
+    -------
+    samples_points(numpy.ndarray)
+        A numpy array representing sampled points with dim = [num_points, 3]
+    (if True) samples_points_normals(numpy.ndarray)
+        A numpy array representing the normal vector of sampled points  with dim = [num_points, 3]
+
+    Examples
```

Comment:
this is a bit hard to digest. consider splitting it up in a few steps with comments
```rst
Examples
--------
Make a triangle mesh.
>>> ...

Compute sample points.
>>> ...

Visualize with Matplotlib.
>>> ...